test/nodes/test_shaper.py

```diff
@@ -0,0 +1,673 @@
+    with pytest.raises(Exception, match="Invalid function arguments") as e:
+        pipeline.run(
+            query="What can you tell me about Berlin?",
+            documents=[Document("Berlin is an amazing city."), Document("I love Berlin.")],
+        )
+
+
+def test_basic_function_batch_invocation(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+            - name: shaper
+              params:
+                inputs:
+                    query:
+                      func: expand
+                      output: questions
+                      params:
+                        expand_target: query
+                        size:
+                          func: len
+                          params:
+                            - documents
+                    documents:
+                      func: concat_docs
+                      output: documents
+                      params:
+                        docs: documents
+                        delimiter: " "
+              type: Shaper
+            pipelines:
+              - name: query
+                nodes:
+                  - name: shaper
+                    inputs:
+                      - Query
+        """
+        )
+    pipeline = Pipeline.load_from_yaml(path=tmp_path / "tmp_config.yml")
+    result = pipeline.run_batch(
+        queries="What can you tell me about Berlin?",
+        documents=[Document("Berlin is an amazing city."), Document("I love Berlin.")],
+    )
+    assert result
+    assert "queries" in result and "query" in result
+    assert "questions" in result["meta"]["invocation_context"]
+
+    result = pipeline.run_batch(
+        queries=["What can you tell me about Berlin?", "Is Berlin as cool as they say?"],
+        documents=[
+            [Document("Berlin is an amazing city."), Document("I love Berlin.")],
+            [Document("Berlin is a wonderful city."), Document("Berlin is pretty cool.")],
+        ],
+    )
+    assert result
+    assert "queries" in result
+    assert len(result["meta"]) > 0
+    assert "questions" in result["meta"][0]["invocation_context"]
+
+
+def test_prompt_node_with_shaper(tmp_path):
+    # tests that the prompt node works with the shaper node
+    # here we use shaper to expand the query to the size of documents and rename the output to questions
+    # this use case was the original motivation for the introduction of the shaper
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+              - name: pmodel
+                type: PromptModel
+              - name: shaper
+                params:
+                  inputs:
+                    query:
+                      func: expand
+                      output: questions
```

Comment:
The way I understand the current implementation, if we would set the output to "query", it would be written to `meta["invocation_context"]["query"]` and if I understand this line correctly: https://github.com/deepset-ai/haystack/blob/6af4f14fe0d375a1ae0ced18930a9239401231c7/haystack/nodes/prompt/prompt_node.py#L885 we would then get an error because we would pass the same kwarg multiple times. Am I wrong?